templates/collection.yaml

```diff
@@ -654,13 +654,28 @@ Resources:
                 - events.amazonaws.com
             Action:
               - 'sts:AssumeRole'
-  EventBridgeCollectAll:
+  EventBridgeCollectNonCloudTrail:
     Type: 'AWS::Events::Rule'
     DependsOn:
       - DeliveryStreamPolicy
     Properties:
-      Description: Export all events
-      EventPattern: !Sub '{ "account": [ "${AWS::AccountId}" ] }'
+      Description: Export all non-CloudTrail events
+      EventPattern: '{ "detail-type": [{ "anything-but": "AWS API Call via CloudTrail" }] }'
+      Targets:
+        - Id: ObserveKinesisFirehose
+          Arn: !GetAtt DeliveryStream.Arn
+          RoleArn: !GetAtt EventBridgeRole.Arn
+  EventBridgeCollectCloudTrail:
+    Type: 'AWS::Events::Rule'
+    DependsOn:
+      - DeliveryStreamPolicy
+    Properties:
+      Description: Export CloudTrail events
+      EventPattern: !Sub
+        - '{ "detail-type": ["AWS API Call via CloudTrail"], "detail": {"eventSource": [{ "anything-but": ["${sources}"] }] }  }'
```

Comment:
`anything-but` takes an array of strings:

https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-event-patterns-content-based-filtering.html#eb-filtering-anything-but

This is rendering a single-item array with comma-separated separated values in that entry, which doesn't seem like it will work for anything other than one source.

Seems like you would need this:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ToJsonString.html

Reply:
this is the same pattern used by EventBridgeSnapshotConfig

Reply:
i think it's a bit confusing so I'll test out Fn::ToJsonString

Reply:
i don't think i want to enable AWS::LanguageExtensions for the entire template so i'll keep things this way.

the pattern I am using ends up leading to multiple elements in the array